plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java

```diff
@@ -118,11 +119,18 @@ private T tryConstructor(Constructor<T> constructor, Mapping config, Configurati
                         final Type pt = parameters[i].getParameterizedType();
                         final Configurator lookup = context.lookupOrFail(pt);
 
-                        final ArrayList<Object> list = new ArrayList<>();
+                        final Collection<Object> collection;
+
+                        if (Set.class.isAssignableFrom(parameters[i].getType())) {
```

Comment:
does it break without this change? surprised you needed to change this

Reply:
Actually was wondering the same thing but ended up approving the PR instead of asking...
![](https://media.giphy.com/media/NGV4vAghFiUOA/giphy.gif)

Reply:
Yes, otherwise I get an argument mismatch when configuring the instance using `@ConfiguredWithCode`.